src/docker.cc

```diff
@@ -46,6 +46,21 @@ constexpr const char kDockerContainerResourcePrefix[] = "container";
 DockerReader::DockerReader(const MetadataAgentConfiguration& config)
     : config_(config), environment_(config) {}
 
+bool DockerReader::ValidateConfiguration() const {
+  try {
+    const std::string container_filter(
+        config_.DockerContainerFilter().empty()
+        ? "" : "&" + config_.DockerContainerFilter());
+
+    QueryDocker(std::string(kDockerEndpointPath) + "/json?all=true&limit=1" + container_filter);
```

Comment:
Careful -- `config_.DockerContainerFilter()` may already include a `limit`. Should investigate which one takes precedence -- you may have to move your `&limit=1` to the end.

Reply:
Great catch, I did some local testing and it appears that the docker server discards the second value, when two limits are sent in the URL. We'll leave it as-is for now.

Reply:
Let's add a comment here...

Reply:
Done.